`valid-expect` reports "unknown modifier" for `expect.assert.*` member calls (e.g. `expect.assert.isDefined`)

### Describe the bug

`valid-expect` reports `Expect has an unknown modifier` for member calls on `expect.assert`, such as `expect.assert.isDefined(x)` or `expect.assert.exists(x)`.

`expect.assert` is a documented Vitest API ([docs](https://vitest.dev/api/expect.html#assert)) — it re-exports chai's `assert` on `expect`, including all of its member methods (`isDefined`, `exists`, `strictEqual`, ...). The bare call form `expect.assert(x)` is correctly accepted; only the member-call form is flagged, presumably because `assert` is parsed as a modifier chain (like `not` / `resolves`) instead of being recognized as a utility member.

This looks like the same class of issue as #921 (`expect.soft`), #922 (`expect.poll` / `expect.element`), and #924 (`expect.any`).

### Reproduction

```ts
import { expect, test } from 'vitest';

function find(): string | undefined {
  return 'value';
}

test('narrows with expect.assert', () => {
  const row = find();
  expect.assert(row); // ok, not flagged
  expect.assert.isDefined(row); // ⚠ valid-expect: Expect has an unknown modifier
  expect(row.length).toBeGreaterThan(0);
});
```

```js
// eslint.config.mjs
import vitest from '@vitest/eslint-plugin';
import tseslint from 'typescript-eslint';

export default [
  {
    files: ['**/*.test.ts'],
    languageOptions: { parser: tseslint.parser },
    plugins: { vitest },
    rules: { 'vitest/valid-expect': 'warn' },
  },
];
```

```
$ eslint example.test.ts
  10:3  warning  Expect has an unknown modifier  vitest/valid-expect
```

### Expected behavior

No warning — `expect.assert.*` calls are valid Vitest usage.

### Versions

- `@vitest/eslint-plugin`: 1.6.24
- `eslint`: 10.8.0
- `vitest`: 4.1.8

For reference, the oxlint port of this rule inherits the same behavior (reproduced on oxlint 1.76.0), so a fix here will eventually propagate there as well.

--- a/src/utils/parse-vitest-fn-call.ts
+++ b/src/utils/parse-vitest-fn-call.ts
@@ -308,6 +308,9 @@ const classifyExpectChain = (
     return { kind: 'matcher', member, callExpression }
   }
 
+  if (isFirstMemberOfStaticExpectCall && name === 'assert')
+    return { kind: 'expect-api', member, callExpression }
+
   if (chaiChainableProperties.has(name))
     return { kind: 'language-chain', member, callExpression }
 

--- a/tests/valid-expect.test.ts
+++ b/tests/valid-expect.test.ts
@@ -5,6 +5,9 @@ ruleTester.run(rule.name, rule, {
   valid: [
     'expect.hasAssertions',
     'expect.hasAssertions()',
+    'expect.assert(value)',
+    'expect.assert.isDefined(value)',
+    'expect.assert.exists(value)',
     'expect.any(String)',
     'expect.anything()',
     'expect.arrayContaining([1])',

